swarm/pss/client/client.go

```diff
@@ -20,44 +19,32 @@ import (
 const (
 	inboxCapacity  = 3000
 	outboxCapacity = 100
-	defaultWSHost  = 8546
 	addrLen        = common.HashLength
 )
 
-// RemoteHost: hostname of node running websockets proxy to pss (default localhost)
-// RemotePort: port of node running websockets proxy to pss (0 = go-ethereum node default)
-// Secure: whether or not to use secure connection
-// SelfHost: local if host to connect from
-type ClientConfig struct {
-	SelfHost   string
-	RemoteHost string
-	RemotePort int
-	Secure     bool
-}
+// After a successful connection with Client.Start, BaseAddr contains the swarm overlay address of the pss node
+type Client struct {
+	BaseAddr []byte
 
-func NewClientConfig() *ClientConfig {
-	return &ClientConfig{
-		SelfHost:   "localhost",
-		RemoteHost: "localhost",
-		RemotePort: 8546,
-	}
-}
+	// peers
+	peerPool map[pss.Topic]map[pot.Address]*pssRPCRW
+	protos   map[pss.Topic]*p2p.Protocol
 
-type Client struct {
-	localuri     string
-	remoteuri    string
-	ctx          context.Context
-	cancel       func()
-	subscription *rpc.ClientSubscription
-	topicsC      chan []byte
-	msgC         chan pss.APIMsg
-	quitC        chan struct{}
-	ws           *rpc.Client
-	lock         sync.Mutex
-	peerPool     map[pss.Topic]map[pot.Address]*pssRPCRW
-	protos       map[pss.Topic]*p2p.Protocol
+	// rpc connections
+	rpc *rpc.Client
+	sub *rpc.ClientSubscription
+
+	// channels
+	topicsC chan []byte
+	msgC    chan pss.APIMsg
+	quitC   chan struct{}
+
+	ctx    context.Context
```

Comment:
The context should not be a struct field, all methods which require a context should accept it as a first argument, usually named `ctx`.

The main reason is that if I pass a context with a timeout, I want that timeout to only affect the immediate method I am calling, not affect all future calls on an object.

See https://golang.org/pkg/context/#pkg-overview for reference.